server/src/main/java/org/elasticsearch/cluster/routing/RoutingNodes.java

```diff
@@ -1307,33 +1307,29 @@ private void ensureMutable() {
      * all the nodes have been returned.
      */
     public Iterator<ShardRouting> nodeInterleavedShardIterator() {
-        final Queue<Iterator<ShardRouting>> queue = new ArrayDeque<>();
-        for (Map.Entry<String, RoutingNode> entry : nodesToShards.entrySet()) {
-            queue.add(entry.getValue().copyShards().iterator());
+        final Queue<Iterator<ShardRouting>> queue = new ArrayDeque<>(nodesToShards.size());
```

Comment:
I wonder, can't we just get the same thing we have here by enhancing `org.elasticsearch.common.collect.Iterators#concat` to `null` out the iterators in its internal array once done? Seems like that thing already does what we do here and does so somewhat more efficiently even I'd say?

Reply:
🤦 never mind, another stupid question :)